spec/lib/workos/organizations_spec.rb

```diff
@@ -6,16 +6,80 @@
 
   describe '.create_organization' do
     context 'with valid payload' do
-      it 'creates an organization' do
-        VCR.use_cassette 'organization/create' do
-          organization = described_class.create_organization(
-            domains: ['example.io'],
-            name: 'Test Organization',
-          )
+      context 'with no idempotency key' do
+        it 'creates an organization' do
+          VCR.use_cassette 'organization/create' do
+            organization = described_class.create_organization(
+              domains: ['example.io'],
+              name: 'Test Organization',
+            )
 
-          expect(organization.id).to eq('org_01FCPEJXEZR4DSBA625YMGQT9N')
-          expect(organization.name).to eq('Test Organization')
-          expect(organization.domains.first[:domain]).to eq('example.io')
+            expect(organization.id).to eq('org_01FCPEJXEZR4DSBA625YMGQT9N')
+            expect(organization.name).to eq('Test Organization')
+            expect(organization.domains.first[:domain]).to eq('example.io')
+          end
+        end
+      end
+
+      context 'with idempotency key' do
+        context 'when idempotency key is used once' do
+          it 'creates an organization' do
+            VCR.use_cassette 'organization/create_with_idempotency_key' do
+              organization = described_class.create_organization(
+                domains: ['example.io'],
+                name: 'Test Organization',
+                idempotency_key: 'key',
+              )
+
+              expect(organization.name).to eq('Test Organization')
+              expect(organization.domains.first[:domain]).to eq('example.io')
+            end
+          end
+        end
+
+        context 'when idempotency key is used more than once' do
+          context 'with duplicate event payloads' do
+            it 'creates an organization' do
```

Comment:
I'm looking into this one, but idempotency keys expire after 24 hours: Idempotency keys expire after 24 hours, so we generate a new request if you reuse a key outside of that time frame.

I'm trying to think of the best way to test this from the SDK.

Reply:
I was thinking, create an org, use something like [Timecop](https://github.com/travisjeffery/timecop) to move ahead by more than 24 hours, then make sure the same payload will create a new organization, even if the same idempotency key was given... Haven't tried any of that 😓, but the general idea should hopefully work?

Reply:
ooh I'll check that out and see, thanks!

Reply:
I think a test like that would make more sense in the API test suite, since all of our SDK tests are just using mocked requests it really won't be asserting anything useful unfortunately.

Reply:
@ams11 is it ok to skip this test per Rob's comment above? Otherwise, I can work on it today. Thanks!

Reply:
Ahh, that's a good point, thanks @robframpton! And agreed, no need to test our mocked requests here, so 👍 to skipping this test, sorry about the wild goose chase over here, @blairlunceford!